### PR TITLE
[TRAFODION-2596] Improve the log4j and log4cxx infrastructure in Traf…

### PR DESCRIPTION
**Pull request by selvaganesang**
…odion

The following changes are made in the way Trafodion logs the messages:

Writes to a log file based on a component or set of components.

C++ part of the code base:

Component | Default log file |  Configuration File
--------------|------------------|--------------------
TM  |  tm_\<nid\>.log   | log4cxx.trafodion.tm.config
SSCP |   sscp_\<nid\>.log   |   log4cxx.trafodion.sscp.config
SSMP |   ssmp_\<nid\>.log  |   log4cxx.trafodion.ssmp.config
All SQL processes (mxosrvr, sqlci, tdm_arkesp, tdm_arkcmp, tdm_udrserv) |  trafodion.sql_\<nid\>.log |    log4cxx.trafodion.sql.config

Java part of the code base

Component  | Default log file |  Configuration File
--------------|------------------|--------------------
TM  |   trafodion.dtm.log  |  log4j.dtm.config
SQL  |  trafodion.sql.java.log   |  log4j.sql.config

By default, the log level is set to INFO for most of the cases. When a message
dominates the log file and if doesn't add value in the current level, it will be
changed to  the higher level in the hierarchy. This should help to improve the
usefulness of the log file at the default INFO level.

The existence of an environment variable TRAF_MULTIPLE_SQL_LOG_FILE will revert back
to the old way of logging into multiple files. Then, the configuration file
log4cxx.trafodion.masterexe.config will be used.

Currently, RollingFileAppender appender is used for both modes. Different config files
are used to change this appender when the need arises.

log4cxx.trafodion.udr.config and log4cxx.trafodion.lob.config are removed.
log4j.hdfs.config used as the config file for SQL is renamed to log4j.sql.config

Foundation components logging will be revamped later.

### CODE REVIEW

File: core/conn/odbc/src/odbc/nsksrvr/SrvrMain.cpp [hunk 1]
```diff
@@ -246,8 +246,20 @@ catch(SB_Fatal_Excep sbfe)
 	myProcName = proc_info.process_name;
 
 	char logNameSuffix[32];
-	sprintf( logNameSuffix, "_%d_%d.log", myNid, myPid );
-	CommonLogger::instance().initLog4cxx("log4cxx.trafodion.masterexe.config", logNameSuffix);
+        const char *lv_configFileName = "log4cxx.trafodion.sql.config";
```
Comment (DaveBirdsall) [inline core/conn/odbc/src/odbc/nsksrvr/SrvrMain.cpp]: Thanks for keeping the old behavior as an option. That will make debugging easier when one wishes to attend to just one process.

**Comment by selvaganesang** [comment]
@arvind-narain Thanks for your input.  Please review this PR

**Comment by arvind-narain** [comment]
Thanks Selva for this change and for taking care of the SQL.HBASE error in monitor logs. 

Would the following also need to be touched ? dbsecurity/auth/src/authEvents.cpp ? logging in the dbsecurity library was implemented by @robertamarton.

**Comment by arvind-narain** [comment]
+1 for this change.

Also have noticed the following additional errors in the logs since Apr 20th - when TRAFODION-2594 went in. Possible these were not getting logged earlier.  

`2017-04-21 19:44:47,715, ERROR, SQL.HBase, Node Number: 0, CPU: 0, PIN: 16166, Process Name: $Z000D6W,,,Java exception in file ../executor/HBaseClient_JNI.cpp, line 2404.
2017-04-21 19:44:47,716, ERROR, SQL.HBase, Node Number: 0, CPU: 0, PIN: 16166, Process Name: $Z000D6W,,,HBaseClient_JNI::getHbaseTableInfo() error: java.lang.ArrayIndexOutOfBoundsException: -3
org.trafodion.sql.HBaseClient.getHbaseTableInfo(HBaseClient.java:1498).
2017-04-21 19:44:49,265, ERROR, SQL.HBase, Node Number: 0, CPU: 0, PIN: 16166, Process Name: $Z000D6W,,,Java exception in file ../executor/HBaseClient_JNI.cpp, line 2404.
2017-04-21 19:44:49,265, ERROR, SQL.HBase, Node Number: 0, CPU: 0, PIN: 16166, Process Name: $Z000D6W,,,HBaseClient_JNI::getHbaseTableInfo() error: java.lang.ArrayIndexOutOfBoundsException: -1
org.trafodion.sql.HBaseClient.getHbaseTableInfo(HBaseClient.java:1498).

====

2017-04-21 19:45:43,720, ERROR, SQL.HBase, Node Number: 0, CPU: 0, PIN: 16224, Process Name: $Z000D8J,,,HBaseClient_JNI::checkAndDeleteRow() error: .
2017-04-21 19:45:43,805, ERROR, SQL.HBase, Node Number: 0, CPU: 0, PIN: 16224, Process Name: $Z000D8J,,,HBaseClient_JNI::checkAndDeleteRow() error: .
2017-04-21 19:45:43,822, ERROR, SQL.HBase, Node Number: 0, CPU: 0, PIN: 16224, Process Name: $Z000D8J,,,HBaseClient_JNI::checkAndDeleteRow() error: .

====`

**Comment by arvind-narain** [comment]
One other comment (last one :) ) - any script that expects master_exec logs in the logs directory ( like sqtools.sh) will need to be modified to pick up the new log file name.

**Comment by selvaganesang** [comment]
Thanks Arvind. I didn't change the security logging because the security logging might have different kind of requirements. I will let @robertamarton  to manage it after this change is merged.

**Comment by selvaganesang** [comment]
It looks like the messages are seen in your 2nd comment are seen in the log because the log level for  SQL.Hbase category is changed to INFO.  Can you please file JIRAs so that it can be picked up the relevant contributors.

**Comment by arvind-narain** [comment]
+1 Thanks Selva for handling the extra error messages in the logs and sqtools.sh

**Comment by selvaganesang** [comment]
jenkins, retest

**Pull request closed without merging**